business/services.go

```diff
@@ -105,13 +105,19 @@ func (in *SvcService) buildServiceList(namespace models.Namespace, svcs []core_v
 	validations := in.getServiceValidations(svcs, deployments, pods)
 	// Convert each k8s service into our model
 	for i, item := range svcs {
-		sPods := kubernetes.FilterPodsForService(&item, pods)
-		/** Check if Service has istioSidecar deployed */
-		mPods := models.Pods{}
-		mPods.Parse(sPods)
-		hasSidecar := mPods.HasIstioSidecar()
```

Comment:
Perhaps the same goal can be achieved using the pods.
If you see the caller "buildServiceList" is invoked from the parent "GetServiceList" which already is fetching pods + deployments for validations.
I suspect that perhaps the deployments in that level can be changed per a generic workloads.
But if we don't want to refactor that in this work, perhaps the mPods.HasIstioSidecar() may have this logic to define if pods have sidecar.
Otherwise I see that in the caller we fetch pods + deployments and in the inner "buildServiceList" we fetch the workloads again that internally will pull also similar objects, even cached, but it may introduce some redundance here.

In the pods we can check if there is Istio sidecar available looking into the containers and if necessary some labeling I think.

Reply:
This is the change (sorry @jmazzitelli for the screenshot), I guess:
![image](https://user-images.githubusercontent.com/1662329/122051040-6ffcad00-cde4-11eb-8800-2e6d281351ea.png)

This what we could tweak, to check if for a service, if there is a pod with an Istio sidecar could be enough to collect Service Health.

Main issue in https://github.com/kiali/kiali/issues/3904 is that when there are services with workloads in the mesh and out of the mesh, it doesn't show the "service health" (rate traffic) and if at least it should be one, that health should be shown.

As that method is used at workload level, perhaps you can add a flag or another method to differentiate, better to add a little of duplication if the logic responds to two difference scenarios (no problem, in that sense).

Reply:
I was thinking about using "mPods.HasIstioSidecar()", but figured our that in a case when workload has 1+ pods, or saying that there is not 1-to-1 mapping between workloads and pods of service, that solution can fail.
So decided to create another method for workloads sidecars.
Yes I agree that this makes additional backend call to load workloads per services but did not find a way to skip above mentioned potential problem.

Reply:
But, why don't create a 
```
func (pods Pods) HasServiceIstioSidecar() bool {
...
}
```

Or similar with the check that if at least one pod has the sidecar, then the service can be considered that it has sidecars ?

I guess that is more or less what the PR is trying to do, or perhaps I'm missing some edge ?

Reply:
Added new method HasAnyIstioSidecar which checks if any of pods have sidecar. No additional backend call is done.